Modules/TRD/src/TrackingTask.cxx

```diff
@@ -0,0 +1,306 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TrackingTask.cxx
+/// \author Salman Malik
+///
+
+#include "QualityControl/QcInfoLogger.h"
+#include "DataFormatsTRD/TrackTRD.h"
+#include "DataFormatsTRD/TrackTriggerRecord.h"
+#include "DataFormatsTRD/Constants.h"
+#include "DataFormatsTRD/Tracklet64.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "DataFormatsTRD/CalibratedTracklet.h"
+#include "TRDQC/Tracking.h"
+#include "TRD/TrackingTask.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include <Framework/DataRefUtils.h>
+#include <Framework/DataSpecUtils.h>
+#include "CCDB/BasicCCDBManager.h"
+#include "TLine.h"
+#include "TCanvas.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+using namespace o2::trd::constants;
+
+namespace o2::quality_control_modules::trd
+{
+
+void TrackingTask::retrieveCCDBSettings()
+{
+  if (auto param = mCustomParameters.find("ccdbtimestamp"); param != mCustomParameters.end()) {
+    mTimestamp = std::stol(mCustomParameters["ccdbtimestamp"]);
+    ILOG(Debug, Devel) << "configure() : using ccdbtimestamp = " << mTimestamp << ENDM;
+  } else {
+    mTimestamp = o2::ccdb::getCurrentTimestamp();
+    ILOG(Debug, Devel) << "configure() : using default timestam of now = " << mTimestamp << ENDM;
+  }
+}
+
+void TrackingTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Debug, Devel) << "initialize TRD TrackingTask" << ENDM;
+
+  retrieveCCDBSettings();
+  buildHistograms();
+}
+
+void TrackingTask::startOfActivity(Activity& activity)
+{
+  ILOG(Debug, Devel) << "startOfActivity " << activity.mId << ENDM;
+}
+
+void TrackingTask::startOfCycle()
+{
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+}
+
+void TrackingTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  ILOG(Debug, Devel) << "monitorData" << ENDM;
+
+  auto trackqcArr = ctx.inputs().get<gsl::span<o2::trd::TrackQC>>("tracksqc");
+  auto tracktrigArr = ctx.inputs().get<gsl::span<o2::trd::TrackTriggerRecord>>("trigrectrk");
+
+  for (const auto& tracktrig : tracktrigArr) {
+    int first = tracktrig.getFirstTrack();
+    int end = first + tracktrig.getNumberOfTracks();
+    // count number of matched tracks
+    int nMatched = 0;
+
+    for (int itrack = first; itrack < end; ++itrack) {
+      auto& trackqc = trackqcArr[itrack];
+      // match only ITS-TPC tracks
+      auto trackType = trackqc.refGlobalTrackId.getSource();
+      if (!(trackType == GTrackID::Source::ITSTPC)) {
+        continue;
+      }
+      nMatched++;
+      // remove tracks with pt < 1 GeV/c
+      if (trackqc.trackTRD.getPt() < 1) {
+        continue;
```

Comment:
I would not put a hard cut here. Can you make the lower pT threshold configurable from json?

Reply:
please stick to the naming convention for class member variables `mPtMin` and adjust the comment in the line above

Reply:
hi @martenole . thanks. i changed the variable name and the comment. thanks